.github/workflows/main.yml

```diff
@@ -0,0 +1,36 @@
+name: Main Workflow
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  main:
+    name: Main Job
+    runs-on: macos-12
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache Mint packages
+        id: mint-cache-v2
+        uses: actions/cache@v3
+        with:
+          path: ~/.mint
+          key: ${{ runner.os }}-mint-${{ hashFiles('Mintfile') }}
+          restore-keys: ${{ runner.os }}-mint-
+
+      - name: Install Mint
+        run: brew install mint
+
+      - name: Install Mint Packages
+        if: steps.mint-cache-v2.outputs.cache-hit != 'true'
+        run: mint bootstrap
```

Comment:
This is a bit "tricky" part. We have set up the linting & formatting to install dependencies via Mint. Now we need to install mint on CI to resolve the dependencies.

How Mint works, it downloads the dependencies and builds an executable Swift package out of the source codes. In order to not perform the building every time, we set up caching on the Mint directory and only call `Install Mint Packages` if the cache is not available.

The cache:
- `path`: which directory to cache
- `key`: the key that should indicate if the cache is valid or not. You can see it hashes the Mintfile where we are locking the versions of the dependencies. If no change in the Mintfile => we can re-use the latest dependencies
- `restore-keys`: basically a fall-back `key` when there is a mismatch in e.g. Mintfile to use caches that are matching at least the beginning of the key to call bootstrap from the latest good cache

The cache is automatically caching before finishing up the workflow.

We only use the `id` of the cache step to reference in the `Install Mint Packages` the result of the caching to be able to skip install if there were caches.
|Without cache hit|With cache hit|
|-|-|
|<img width="1042" alt="Screen Shot 2022-05-02 at 1 03 41 AM" src="https://user-images.githubusercontent.com/1413987/166168178-b493b47c-1a54-4c83-ae34-2ee032252923.png">|<img width="1046" alt="Screen Shot 2022-05-02 at 1 03 12 AM" src="https://user-images.githubusercontent.com/1413987/166168180-96f086d3-bdf1-450a-bf88-c7cbe685bd71.png">|